chrome/content/zotero/xpcom/zotero.js

```diff
@@ -381,7 +381,26 @@ Components.utils.import("resource://gre/modules/osfile.jsm");
 					return false;
 				}
 			}
-			
+			if(Zotero.File.isDropboxDirectory(dataDir.path) && Zotero.Prefs.get('warnDropboxDataDir')){
+				let check = {value: false};
+				ps = Services.prompt;
+				index = ps.confirmEx(null,
```

Comment:
Correct, but all declarations are inside conditionals, so conceivably it may be possible for it to end up undeclared. Also, if something is declared within a block, removing that block should have no effect on that same variable somewhere else. Here it would. It's ok to re-declare variables with `var`, the IDE should only be warning you, not throwing errors. With `let`, it's an error to re-declare the same variable within the same block,  but that wouldn't be an error here either.